### ISSUE TITLE
Failing Plate tests

### ISSUE THREAD

**Issue by yuriy0** [hakaru-dev/hakaru#50]
These tests don't pass (as of 39cb306f), but they should. 

### Don't be confused by extra iterations at beginning of plate

```
 Expected result : Plate(n,i,Gaussian(0,2^(1/2)))
 Evaluated result: Weight(2^(-n)*Pi^(-n),Plate(n,i,Bind(Lebesgue(-infinity,infinity),`xx俏`,piecewise(-1 < i,Weight(exp(-1/4*`xx俏`^2)*Pi^(1/2),Ret(`xx俏`)),Msum()))))
```
The condition `-1 < i` is not being simplified to `true` in the context `0 <= i`. It isn't exactly clear how binders in Maple produced by Plate interact with Domain (if at all) but the context isn't being correctly passed in this case.

### plate size failed
```
 Expected result : Weight(f(k),Ret(Unit))
 Evaluated result: Bind(Plate(k,c,Uniform(37,42)),`xs俹`,Weight(f(Hakaru:-size(`xs俹`)),Ret(Unit)))
```
Probably related to the above; but the information about the size of `xs` would be inserted in a slightly different place.

### gmm (currently failing -- due to alpha-equivalence) failed
The results are equal, although not quite alpha-equivalent (more precisely, they are equal up to commutativity of `+` and `*`, and `sqrt(1/x)=1/sqrt(x)`, and alpha equivalence). 

### gmm total (currently failing -- due to alpha-equivalence) failed
Same as above; and also differs by a constant factor inside a `product`. I'm not sure if these simplifications are very important.

### Conjugacy between unrolled symmetric Dirichlet and multinomial

```
Error, improper op or subscript selector
```
Investigation needed.

### Conjugacy between rolled Dirichlet and multinomial
```
Error, (in Loop:-unproduct) assertion failed, unproduct: BUG! bijectivity assumed!
```
At least it's obvious where this one happens, but it isn't clear if this is a regression in KB or a bug in unproduct (as the assertion claims) that is only now being hit.

**Comment by JacquesCarette** [comment]
For the future: unless you are 100% sure that some failures are related, it is best to open a different issue for each.

**Comment by yuriy0** [comment]
Fixed as of 2221a66 ("Conjugacy between rolled Dirichlet and multinomial" works but has a subtle issue, #56)
